test/framework/src/main/java/org/elasticsearch/test/MockLogAppender.java

```diff
@@ -267,58 +294,57 @@ public String toString() {
         }
     }
 
+    public Releasable capturing(Class<?>... classes) {
+        this.loggers = Arrays.stream(classes).map(Class::getCanonicalName).toList();
+        addToMockAppenders(this, loggers);
+        return this;
+    }
+
+    public Releasable capturing(String... names) {
+        this.loggers = Arrays.asList(names);
+        addToMockAppenders(this, loggers);
+        return this;
+    }
+
     /**
      * Adds the list of class loggers to this {@link MockLogAppender}.
      *
      * Stops and runs some checks on the {@link MockLogAppender} once the returned object is released.
      */
-    public Releasable capturing(Class<?>... classes) {
```

Comment:
I'm confused: you stated in the description that you wanted to replace `capturing` gradually (hence the non-provate `MockLogAppender` ctor), but here you are removing them. What am I missing?

Reply:
They still exist, just above this. The diff shows weird because the javadoc is moved to the new method.

Reply:
Ah! Got it. Yes, the diff was confusing.